src/main/java/com/synopsys/integration/detect/workflow/componentlocationanalysis/ScanResultToComponentListTransformer.java

```diff
@@ -29,44 +28,46 @@ public class ScanResultToComponentListTransformer {
         for (DeveloperScansScanView component : rapidScanFullResults) {
             if (component.getExternalId() == null) {
                 componentNamesWithNullIds.add(component.getComponentName());
             } else {
                 componentIdWithMetadata.put(component.getExternalId(), populateMetadata(component));
+                orderedComponentIDs.add(component.getExternalId());
             }
         }
         for (String componentNameToWarnAbout : componentNamesWithNullIds) {
             logger.warn("Component '{}' is skipped due to missing external ID in the scan result json.", componentNameToWarnAbout);
         }
 
-        return convertExternalIDsToComponentList(componentIdWithMetadata);
+        return convertExternalIDsToComponentList(componentIdWithMetadata, orderedComponentIDs);
     }
 
-    private Set<Component> convertExternalIDsToComponentList(HashMap<String, ScanMetadata> componentIdWithMetadata) {
-        Set<Component> componentSet = new HashSet<>();
+    private Set<Component> convertExternalIDsToComponentList(HashMap<String, ScanMetadata> componentIdWithMetadata, Set<String> orderedComponentIDs) {
+        Set<Component> componentSet = new LinkedHashSet<>();
         try {
-            for (String componentIdString : componentIdWithMetadata.keySet()) {
+            for (String componentIdString : orderedComponentIDs) {
+                JsonObject jsonObject = getJsonObjectFromScanMetadata(componentIdWithMetadata.get(componentIdString));
                 String[] parts;
                 if ((parts = componentIdString.split(":")).length == 3) {
                     // For Maven and Gradle, the componentId is of the form "g:a:v"
-                    componentSet.add(new Component(parts[0], parts[1], parts[2], getJsonObjectFromScanMetadata(componentIdWithMetadata.get(componentIdString))));
+                    componentSet.add(new Component(parts[0], parts[1], parts[2], jsonObject));
                 } else if ((parts = componentIdString.split("/")).length == 2) {
                     // For NPM and NuGet, the componentId looks is of the form "a/v"
-                    componentSet.add(new Component(null, parts[0], parts[1], getJsonObjectFromScanMetadata(componentIdWithMetadata.get(componentIdString))));
+                    componentSet.add(new Component(null, parts[0], parts[1], jsonObject));
                 }
             }
         } catch (Exception e) {
             logger.debug("There was a problem processing component IDs from scan results during Component Location Analysis: {}", e);
         }
         return componentSet;
     }
-
+    
     private JsonObject getJsonObjectFromScanMetadata(ScanMetadata scanMeta) {
         Gson gson = new GsonBuilder().create();
-        JsonElement element = gson.toJsonTree(scanMeta);
-        JsonObject object = element.getAsJsonObject();
+        JsonObject object = gson.fromJson(gson.toJson(scanMeta), JsonObject.class);
         return object;
```

Comment:
Potential for code refractor later(after 8.11) to move this as a getter inside ScanMetadata class.